plugins/module_utils/users.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2024, Jeff Kala (@jeffkala)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible_collections.networktocode.nautobot.plugins.module_utils.utils import (
+    NautobotModule,
+    ENDPOINT_NAME_MAPPING,
+)
+
+
+NB_USERS = "users"
+NB_ADMIN_GROUP = "groups"
+NB_OBJECT_PERMISSION = "permissions"
+
+
+class NautobotUsersModule(NautobotModule):
+    def run(self):
+        """
+        This function should have all necessary code for endpoints within the application
+        to create/update/delete the endpoint objects
+        Supported endpoints:
+        users
+        groups
+        permissions
+        """
+        # Used to dynamically set key when returning results
+        endpoint_name = ENDPOINT_NAME_MAPPING[self.endpoint]
+
+        self.result = {"changed": False}
+
+        application = self._find_app(self.endpoint)
+        nb_app = getattr(self.nb, application)
+        nb_endpoint = getattr(nb_app, self.endpoint)
+        user_query_params = self.module.params.get("query_params")
+
+        data = self.data
+
+        # Used for msg output
+        if data.get("name"):
+            name = data["name"]
```

Comment:
I think you're going to need `name = data["username"]` for the admin_user module here.